.github/workflows/run-build-and-start-check.yml

```diff
@@ -0,0 +1,58 @@
+name: Build and start
+
+on:
+  push:
+    branches:
+      - 'main'
+    tags:
+      - 'v*'
+
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  build-and-start:
+    services:
+      postgres:
+        image: postgres
+        env:
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_USER: codex
+          POSTGRES_DB: notes
+        ports:
+            - 5432:5432
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+
+    name: Build and start the app
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Change database address
+        run: |
+          sed -i "s/@postgres/@localhost/" app-config.yaml
+
+      - name: Setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 20
+
+      - name: Install dependencies
+        run: yarn install
+
+      - name: Build
+        run: yarn build
```

Comment:
why compiler can't detect missed `.js` extensions?

Reply:
According to this issue: https://github.com/microsoft/TypeScript/issues/42813 ,

We can try and change compiler's options `module` and `moduleResolution` to `NodeNext` in the `tsconfig.json` file. I tried this locally and it seems to be detecting these falsy imports:
![image](https://github.com/codex-team/notes.api/assets/59017579/f5bde943-8a67-4ab4-a44b-6156ecea68e3)

![image](https://github.com/codex-team/notes.api/assets/59017579/f08c1b60-13c2-4319-a4f7-6e5e40232ce2)

However, I'm not a typescript expert, so this change might break something occasionally. I think it might be for a separate Issue.

Reply:
according to the error on the screenshot, it should also work with node16. and `"moduleResolution": "node16"` is the recommended setting for nodejs20: https://github.com/tsconfig/bases/blob/main/bases/node20.json#L14

so i suggest to set node16 and see if it works

Reply:
Yeah, you are right. The result with node16 is the same.

btw IDE starts to mark these imports as falsy too. That's good
![image](https://github.com/codex-team/notes.api/assets/59017579/04817b19-16e4-4848-9fcd-8ffb5173e72f)

Reply:
I'll change the resolution and fix [36] imports in the next PR then. And see if anything breaks